code/src/Components/NameInput.js

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+export const NameInput = ({ id, name, setName }) => {
+  
+  return (
+      <label className="name-label" htmlFor={id}>
+        1. What's your name?
+        <input 
+          className="name-input"
+          id={id}
+          type="text"
+          onChange={event => setName(event.target.value)}
```

Comment:
An alternative approach for this event-handler could be to create a new function, and in that function we in turn define that our state should be updated/executed with a value (your input in this case) as we did at the last wednesday lecture with Maks (Week 10, Lecture2 21/10).

(@EmOlsz feel free to correct me if I'm wrong! 😅 But it's more clean/common practice to create another function for this, and not pass set state-handlers directly inside our events in JSX? )